src/makeInfinityAdder.js

```diff
@@ -1,12 +1,35 @@
 'use strict';
 
 /***
- *
  * @return {function}
  */
-
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(value) {
+    if (arguments.length === 0) {
+      return sum;
+    }
+
+    sum += value;
+
+    function innerAdder(nextValue) {
+      if (arguments.length === 0) {
```

Comment:
The use of arguments.length to check for no arguments is correct, but it would be more readable and modern to use default parameters to handle the case where no arguments are passed.